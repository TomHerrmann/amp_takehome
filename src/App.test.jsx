import React from 'react';
import { Provider, useStore } from 'react-redux';
import { act } from 'react-dom/test-utils';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AppContainer from './App.jsx';
import {
  apiError,
  contactsPopulate,
  contactUpdateCancel,
} from './actions/actions';
import store from './__testHelpers__/mockStore';

describe('<AppContainer />', () => {
  let state;
  beforeEach(() => {
    state = store.getState();
  });

  it('should render correctly & dispatch contents populate', () => {
    const component = mount(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should correctly fetch all contacts', () => {});
  it('should handle when contact fetch fails', () => {});
  it('should render correctly if detailsView is true', () => {});
  it('should render the loading spinner if isLoading is true', () => {});
  it('should dispatch contentsUpdateCancel on header button onClick', () => {});
});
