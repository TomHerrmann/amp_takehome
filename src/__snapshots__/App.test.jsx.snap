// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer /> should render correctly & dispatch contents populate 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      apiError={[Function]}
      contactIndex={null}
      contactSelect={[Function]}
      contactUpdateCancel={[Function]}
      contactUpdateSave={[Function]}
      contacts={
        Array [
          Object {
            "address": "42991 Beahan Parkway",
            "city": "Veumberg",
            "country": "Venezuela",
            "email": "Marianne56@hotmail.com",
            "firstName": "Brandy",
            "id": "9367",
            "lastName": "Bartell",
            "phone": "586-723-4824",
          },
          Object {
            "address": "356 Marlene Fork",
            "city": "East Preston",
            "country": "Lebanon",
            "email": "Mariana.Rohan@hotmail.com",
            "firstName": "Ines",
            "id": "92819",
            "lastName": "Steuber",
            "phone": "563-954-3264",
          },
        ]
      }
      contactsPopulate={[Function]}
      currentContact={null}
      detailsView={false}
      errorStatus={null}
      isLoading={true}
    >
      <main
        className="app"
      >
        <header
          className="title-container"
        >
          <section
            className="header-title"
          >
            <h1>
              Contacts
            </h1>
          </section>
        </header>
        <section
          className="contacts-wrapper"
        >
          <LoadingSpinner>
            <div
              className="loading-spinner"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
          </LoadingSpinner>
        </section>
      </main>
    </App>
  </Connect(App)>
</Provider>
`;
